javaagent-exporters/prometheus/src/main/java/io/opentelemetry/javaagent/exporters/prometheus/PrometheusMetricProducerCustomizer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.exporters.prometheus;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.exporters.prometheus.PrometheusCollector;
+import io.opentelemetry.javaagent.spi.exporter.MetricProducerCustomizer;
+import io.opentelemetry.sdk.metrics.export.MetricProducer;
+import io.prometheus.client.exporter.HTTPServer;
+import java.io.IOException;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PrometheusMetricProducerCustomizer} registers {@link MetricProducer} to the ${@link
+ * PrometheusMetricProducerCustomizer}. The collector pulls metrics from the producer when the
+ * Prometheus server scraps metric endpoint.
+ */
+@AutoService(MetricProducerCustomizer.class)
+public class PrometheusMetricProducerCustomizer implements MetricProducerCustomizer {
+  private static final Logger log =
+      LoggerFactory.getLogger(PrometheusMetricProducerCustomizer.class);
+
+  private static final String EXPORTER_NAME = "otel.exporter.prometheus";
+  private static final String PORT_CONF_PROP_NAME = EXPORTER_NAME + ".port";
+  private static final String HOST_CONF_PROP_NAME = EXPORTER_NAME + ".host";
+  private static final String DEFAULT_PORT = "9464";
+  // The empty address equals to any address
+  private static final String DEFAULT_HOST = "0.0.0.0";
+
+  @Override
+  public void configure(MetricProducer producer, Properties config) {
+    PrometheusCollector.newBuilder().setMetricProducer(producer).buildAndRegister();
+    try {
+      String portStr = config.getProperty(PORT_CONF_PROP_NAME, DEFAULT_PORT);
+      String host = config.getProperty(HOST_CONF_PROP_NAME, DEFAULT_HOST);
+      log.info("Creating Prometheus exporter on host: '{}' and port: '{}'", host, portStr);
+      new HTTPServer(host, Integer.parseInt(portStr), true);
```

Comment:
Sorry, but I still not quite understand how `HTTPServer` and `PrometheusCollector` are connected :(

Reply:
Added a comment